src/components/OSCALLoader.js

```diff
@@ -49,6 +49,52 @@ const oscalObjectTypes = {
   },
 };
 
+function updateData(data, editedField, newValue) {
+  if (editedField.length === 1) {
+    const editData = data;
+    editData[editedField] = newValue;
+    return;
+  }
+
+  updateData(data[editedField.shift()], editedField, newValue);
+}
+
+/**
+ *
+ * @param data data that will be passed into the body of the PATCH request, doesn't initially contain the updates
+ * @param update function that will update a state, forcing a re-rendering if the PATCH request is successful
+ * @param editedField path to the field that is being updated
```

Comment:
Should this be a bit more descriptive, maybe something like `editedFieldJsonPath` or  `fieldJsonPath` throughout the changes?